text/0086-sentry-bundler-plugins-api.md

```diff
@@ -0,0 +1,110 @@
+
+Facts to consider when deciding on an API:
+
+- The bump to v2 of unplugin will imply a major bump of the Sentry Webpack plugin and its switch to our unplugin based architecture.
+- We recently introduced debug IDs as a source map resolving mechanism. This results in different needed options needed when running the plugins. Some options must be added, some removed.
+- Releases are not a required parameter anymore for source maps upload.
+
+As an additional thought: The new API should be seen as a "clean-slate" API - it's less about what has been and more about what the optimal API should look like.
+
+# Supporting Data
+
+- Debug ID RFC: https://github.com/getsentry/rfcs/blob/ab0f75567d95e247b5cc3f09c466b7254844e654/text/0081-sourcemap-debugid.md
+- Debug ID changes related to the unplugin based Sentry bundler plugins: https://github.com/getsentry/sentry-javascript-bundler-plugins/issues/191
+
+# Options Considered
+
+The API generally only defines the options passed to the individual plugins. The options are the same for all bundlers.
+
+For the sake of simplicity of this RFC, we will not decide on the JSdoc. All JSdoc below is just to illustrate what the options are used for.
+
+Proposed API (TypeScript typedef):
+
+```ts
+interface Options {
+  org: string; // Previously this was optional and could have been set with `env.SENTRY_ORG` or `.sentryclirc` - for the sake of explicity, we remove these configuration methods
+  project: string; // Previously this was optional and could have been set with `env.SENTRY_ORG` or `.sentryclirc` - for the sake of explicity, we remove these configuration methods
+  authToken: string; // Previously this was optional and could have been set with `env.SENTRY_ORG` or `.sentryclirc` - for the sake of explicity, we remove these configuration methods
+  url?: string; // A way to configure self-hosted instances: unchanged
+  headers?: Record<string, string>; // Headers added to every outgoing network request.
+  debug?: boolean; // default: false - Whether to print debug information.
+  silent?: boolean; // default: false - Whether to print anything at all.
+  errorHandler?: (err: Error) => void; // when provided will swallow errors unless the provided function throws itself - By default any error will stop compilation to abort
+  telemetry?: boolean; // default: true - whether to send telemetry to Sentry
+  disable?: boolean; // default: false - just a quick way to entirely disable the plugin - purely for convenience
+
+  // General configuration for source maps uploading - if not provided, source map uploading will be disabled
+  sourcemaps?: {
+    assets: string | string[]; // globs pointing to the javascript files that should be uploaded to Sentry with their source maps - these are the built assets and not the source files
+    ignore?: string | string[]; // globs to exclude javascript files from being uploaded (will also not upload their source maps)
```

Comment:
Should assets/ignore always be an array?

Reply:
Yeah let's keep this for convenience.